moon/controller_excavator_round2.py

```
@@ -283,12 +566,57 @@ def pursue_volatile(v):
+                            self.set_brakes(True)
+
+                            # reset VSLAM (this should reset self.vslam_valid)
+                            self.tf['vslam']['trans_matrix'] = None
+                            self.vslam_reset_at = None
+                            self.send_request('vslam_reset', vslam_reset_time)
+                            while self.vslam_reset_at is None or self.sim_time - self.vslam_reset_at < timedelta(seconds=3):
+                                try:
+                                    self.wait(timedelta(seconds=1))
+                                except BusShutdownException:
+                                    raise
+                                except Exception as e:
+                                    print(self.sim_time, self.robot_name, "Exception while waiting for VSLAM reset: ", str(e))
+
+
+                            while not self.hauler_ready:
+                                try:
+                                    self.wait(timedelta(seconds=3))
+                                except BusShutdownException:
+                                    raise
+                                except Exception as e:
+                                    print(self.sim_time, self.robot_name, "Exception while waiting for hauler to be ready: ", str(e))
+
+                            self.hauler_ready = False
+
+                            self.send_request('external_command hauler_1 request_true_pose')
+                            try:
+                                self.wait(timedelta(seconds=5)) # wait for pose to update (TODO: wait for a flag instead)
+                            except BusShutdownException:
+                                raise
+                            except Exception as e:
+                                print(self.sim_time, self.robot_name, "Exception while waiting for true pose: ", str(e))
```

just curious what kind of exceptions do you expect here? This construction will take everything including "division by zero", so it can easily hide errors. In the future (if it happens January 2021) I would change it to list of all exceptions you really want to catch ... or make some class inheriting all of them, so that you do not have to repeat it again and again .. and `BusShutdownException` would simply raise automatically

---

yeah, this is more on the quick and dirty side. There are quite a few exceptions that the app can trigger that are irrelevant to this (e.g., hauler may request excavator to pause while it's catching up, VSLAM may be recalculating something which I found is better for the robot to stop while that's happening - well, at this point excavator is already paused so it doesn't care). As I am prototyping new ideas (including new types of exceptions), I didn't want to have to go through all the code and be adding them to some explicit list. I am however adding traceback/stack printout so that I am not missing true problems